engine/source/private/core/asset_manager.cpp

```diff
@@ -0,0 +1,22 @@
+﻿#include "core/asset_manager.hpp"
+
+namespace lumina
+{
+    size_t Asset::nextID = 0;
+
+    void AssetManager::UnloadAssets()
```

Comment:
The only issue i can see happening with the current setup is, if we have an asset that has only a single instance and gets removed and placed a lot (for example an RPG rocket), the asset will be loaded and unloaded all the time. But I'm not sure what would be an appropriate solution for that. But it's also a problem for future us, and not current us 😄

Reply:
We ourselves are responsible for when to UnloadAssets, so we might also have a "Rocket" loaded at all times if that is our main characters weapons. 

And I also think this would be more game focused, right?

Reply:
on the game side, i don't think the user needs to be worries about how the asset manager handles things.

We might eventually add some sanity checks that if an asset is loaded often, we can assume we can keep it in memory. This might prevent longer loading times for loading the asset from disk.